src/libs/SearchUIUtils.ts

```diff
@@ -339,6 +353,14 @@ function getIOUReportName(data: OnyxTypes.SearchResults['data'], reportItem: Sea
     });
 }
 
+function getTransactionViolations(allViolations: OnyxCollection<OnyxTypes.TransactionViolation[]>, transaction: SearchTransaction): OnyxTypes.TransactionViolation[] {
+    const transactionViolations = allViolations?.[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`];
+    if (!transactionViolations) {
+        return [];
+    }
+    return transactionViolations.filter((violation) => !isViolationDismissed(transaction, violation));
+}
```

Comment:
Can you refactor this to only pass the `transactionViolations`? Seems like there is no reason to pass the full collection, this is less performant